binaries/cuprated/src/txpool/incoming_tx.rs

```diff
@@ -184,28 +198,35 @@ async fn handle_incoming_txs(
         // Maybe we should remember these invalid txs for some time to prevent them getting repeatedly sent.
         if let Err(e) = check_tx_relay_rules(&tx, context) {
             tracing::debug!(err = %e, tx = hex::encode(tx.tx_hash), "Tx failed relay check, skipping.");
-
-            continue;
+            if drop_relay_rule_errors {
+                continue;
+            } else {
+                return Err(IncomingTxError::RelayRule(e));
+            }
         }
```

Comment:
`/send_raw_transaction` needs this to check for errors; all previous callers maintain the same behavior.